### PR TITLE
Use cfacter instead of facter if possible

### PR DESCRIPTION
**Pull request by nibalizer**
CFacter is the facter replacement written in C++. It is available from
the puppetlabs repo.

### CODE REVIEW

**Comment by emonty** [comment]
More speed in facts is more better

File: lib/ansible/module_utils/facts.py [hunk 1]
```diff
@@ -2734,12 +2734,16 @@ def get_all_facts(module):
     for (k, v) in facts.items():
         setup_options["ansible_%s" % k.replace('-', '_')] = v
 
-    # Look for the path to the facter and ohai binary and set
+    # Look for the path to the facter, cfacter, and ohai binaries and set
     # the variable to that path.
 
-    facter_path = module.get_bin_path('facter')
```
Comment (bcoca) [inline lib/ansible/module_utils/facts.py]: I think you mean to reverse these

File: lib/ansible/module_utils/facts.py [hunk 2]
```diff
@@ -2734,12 +2734,16 @@ def get_all_facts(module):
...
+    facter_path = module.get_bin_path('cfacter')
+    cfacter_path = module.get_bin_path('facter')
     ohai_path = module.get_bin_path('ohai')
```
Comment (bcoca) [inline lib/ansible/module_utils/facts.py]: I think you meant to reverse these

**Comment by amenonsen** [comment]
Looks sensible, has no effect if `cfacter` is not installed.

**Comment by nibalizer** [comment]
I'm unclear why the tests are failing, does anyone know?

**Pull request merged**